opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+            }
+            training_args = TrainingArguments(
+                per_device_train_batch_size=batch_size,
+                gradient_accumulation_steps=1,
+                fp16=False,
+                **default_args,
+            )
+
+            train_dataloader = DataLoader(
+                train_examples,
+                shuffle=True,
+                batch_size=training_args.per_device_train_batch_size,  # Trains with this batch size.
+            )
+
+            print("Start training with accelerator...")
+            print(f"The number of training epoch are {num_epochs}")
+            print(
+                f"The total number of steps training epoch are {len(train_dataloader)}"
+            )
+
+            optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate)
+            scheduler = get_linear_schedule_with_warmup(
+                optimizer,
+                num_warmup_steps=min(10000, 0.05 * len(train_dataloader)),
+                num_training_steps=num_epochs * len(train_dataloader),
+            )
+
+            accelerator = Accelerator(fp16=training_args.fp16)
+            model, optimizer, train_dataloader, scheduler = accelerator.prepare(
+                model, optimizer, train_dataloader, scheduler
+            )
+            init_time = time.time()
+            total_loss = []
+
+            # TO DO: to add more comments to explain the training epoch
+            for epoch in range(num_epochs):
+                print("Training epoch " + str(epoch) + "...")
+                for step, batch in tqdm(
+                    enumerate(train_dataloader), total=len(train_dataloader)
+                ):
+
+                    batch_q = batch[0]
+                    batch_p = batch[1]
+
+                    out_q = model.tokenize(batch_q)
+                    for key in out_q.keys():
+                        out_q[key] = out_q[key].to(model.device)
+
+                    out_p = model.tokenize(batch_p)
+                    for key in out_p.keys():
+                        out_p[key] = out_p[key].to(model.device)
+
+                    Y = model(out_q)["sentence_embedding"]
+                    X = model(out_p)["sentence_embedding"]
+
+                    XY = torch.exp(torch.matmul(X, Y.T) / (X.shape[1]) ** 0.5)
+                    num = torch.diagonal(XY)
+
+                    den = torch.sum(XY, dim=0)
+                    den1 = torch.sum(XY, dim=1)
+
+                    batch_loss = -torch.sum(torch.log(num / den)) - torch.sum(
+                        torch.log(num / den1)
+                    )
+
+                    accelerator.backward(batch_loss)
+                    optimizer.step()
+                    scheduler.step()
+                    optimizer.zero_grad()
+                    total_loss.append(batch_loss.item())
+
+                    if not step % 500 and step != 0:
+                        plt.plot(total_loss[::100])
+                        plt.show()
```

Comment:
For every 500 steps, we will plot total loss and show it. How many steps user will use generally to train model? Will we show too many plots if steps is large number like 10K? Is it ok to not show this if user don't always need this. If we think it may help user to debug or understand details, I think add debug/verbose mode may help.

Reply:
adding verbose to parameters and ask users to opt in to plot

Reply:
resolved in 
[added save_as_pt function](https://github.com/opensearch-project/opensearch-py-ml/pull/18/commits/094339cca0a561d3ecd4edf74ce803b77e973239)